staging/src/k8s.io/kubectl/pkg/proxy/proxy_server.go

```diff
@@ -195,6 +195,7 @@ func NewServer(filebase string, apiProxyPrefix string, staticPrefix string, filt
 	proxy := proxy.NewUpgradeAwareHandler(target, transport, false, false, responder)
 	proxy.UpgradeTransport = upgradeTransport
 	proxy.UseRequestLocation = true
+	proxy.OverrideRequestHost = true
```

Comment:
@FabiamKramm It seems a real issue when using api behind nginx as instance (I remember to see a k3s issue related to this). Can we add a flag to enable OverrideRequestHost? I agree that would be safe to not enabled by default (at least for now IMO). However, we will need people like @liggitt and others to approve this one.

Reply:
Let me update my comment:
I am good **not** using flags as kubeproxy goal is to override localhost and so far looks sane to avoid any breakage with current code.